### DISCUSSION TITLE
How to retrieve data?

### DISCUSSION THREAD

**Discussion by lalit-suthar-dev** [innoveit/react-native-ble-manager#1185] [Q&A]
Can I retrieve data from the digital hemoglobin meter poc-30 point of care device via Bluetooth using react-native-ble-manager, or do I need to use Android code?

**Comment by enzo69006** [comment]
An example how i read data : 

  ```
function readIfActivated() {
    const serviceUUID = "19b10000-e8f2-537e-4f6c-d104768a1214";
    const characteristicUUID = "19b10001-e8f2-537e-4f6c-d104768a1215";

    BleManager.read(antivolID, serviceUUID, characteristicUUID)
      .then((readData) => {
        console.log("Read data: ", readData);

        const isActivated = readData[0] === 0x01;

        setIsAntivolActivated(isActivated);

        console.log("Antivol activated state: ", isActivated ? "Activated" : "Deactivated");
      })
      .catch((error) => {
        console.log("Error reading antivol state: ", error);
      });
  }
```

**Reply by lalit-suthar-dev** [reply]
@wysixontheflux Is this serviceUUID and characteristicUUID static or dynamic? If they are dynamic, how do I obtain them?

**Comment by aminhaerijs** [comment]
> Can I retrieve data from the digital hemoglobin meter poc-30 point of care device via Bluetooth using react-native-ble-manager, or do I need to use Android code?

Having the same problem. Were you able to read the data with this library or did you use the android code?

**Reply by lalit-suthar-dev** [reply]
@aminhaerijs Yes, I used Android code.

**Reply by aminhaerijs** [reply]
Do you have a solution for iOS as well?

**Reply by lalit-suthar-dev** [reply]
@aminhaerijs Solution is only for Android as the device does not support iOS.
